compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -543,17 +544,26 @@ getHlintDiagnosticsRule =
         let modu = pm_parsed_source pm
         (classify, hint) <- useNoFile_ GetHlintSettings
         let ideas = applyHints classify hint [createModuleEx anns modu]
-        return ([toDiagnostic file i | i <- ideas, ideaSeverity i /= Ignore], Just ())
+        return ([diagnostic file i | i <- ideas, ideaSeverity i /= Ignore], Just ())
     where
-      -- To-do : Improve this.
-      toDiagnostic file i = ideHintText file (T.pack $ show i)
-      ideHintText fp msg = (fp, LSP.Diagnostic {
-         _range = noRange,
-         _severity = Just LSP.DsInfo,
-         _code = Nothing,
-         _source = Just "linter",
-         _message = msg,
-         _relatedInformation = Nothing
+      srcSpanToRange :: HSE.SrcSpan -> LSP.Range
```

Comment:
I would guess we already have this function somewhere else. I don't know where though.

Reply:
We don’t. There are multiple things called `SrcSpan`. This is the one from `haskell-src-exts` which is new.